openfisca_core/parameters.py

```diff
@@ -420,30 +388,61 @@ def __getitem__(self, key):
         else:
             raise KeyError(key)
 
+    def __repr__(self):
+        return os.linesep.join([
+            '-' + indent(repr(bracket))[1:]
+            for bracket in self.brackets
+            ])
+
 
 def _parse_child(child_name, child, child_path):
     if 'values' in child:
         return Parameter(child_name, child, child_path)
     elif 'brackets' in child:
         return Scale(child_name, child, child_path)
+    elif isinstance(child, dict) and all([INSTANT_PATTERN.match(key) for key in child.keys()]):
+        return Parameter(child_name, child, child_path)
     else:
         return ParameterNode(child_name, data = child, file_path = child_path)
 
 
-class ParameterNode(AbstractParameter):
+class ParameterNode(ValidableParameter, DatableParameter):
     """
         A node in the legislation `parameter tree <http://openfisca.org/doc/coding-the-legislation/legislation_parameters.html>`_.
     """
-    _data_type = dict
+
+    _allowed_keys = None  # By default, no restriction on the keys
 
     def __init__(self, name = "", directory_path = None, data = None, file_path = None):
         """
         Instanciate a ParameterNode either from a dict, (using `data`), or from a directory containing YAML files (using `directory_path`).
 
         :param string name: Name of the node, eg "taxes.some_tax".
         :param string directory_path: Directory containing YAML files describing the node.
-        :param string data: Object representing the parameter node. It usually has been extracted from a YAML file.
+        :param dict data: Object representing the parameter node. It usually has been extracted from a YAML file.
         :param string file_path: YAML file from which the `data` has been extracted from.
+
+
+        Building from data:
```

Comment:
Instantiating a parameter from JSON(?) data :

Reply:
It's not really JSON data (JSON is a text format used to serialise Javascript objects). It's a python dict.

Would you prefer `Building from a dict` ?